include/exadg/operators/operator_base.cpp

```diff
@@ -784,11 +798,12 @@ OperatorBase<dim, Number, n_components>::internal_init_system_matrix(
   else /* if (!is_dg && !is_mg)*/
     DoFTools::make_sparsity_pattern(dof_handler, dsp, *this->constraint);
 
-  SparsityTools::distribute_sparsity_pattern(dsp,
-                                             owned_dofs,
-                                             dof_handler.get_communicator(),
-                                             relevant_dofs);
-  system_matrix.reinit(owned_dofs, owned_dofs, dsp, dof_handler.get_communicator());
+  // only work on this aspect if we are part of the communicator with the dofs
+  if(Utilities::MPI::sum(owned_dofs.n_elements(), mpi_comm) == owned_dofs.size())
```

Comment:
seems to be the same expression as above, so maybe introduce a variable, e.g. `bool my_rank_is_part_of_sub_communicator`?

Reply:
Fixed in cc655075

Reply:
I decided to use an additional bool variable as suggested rather than the initial size variable, because it allows me to skip the comment. See af04232b